a1/quiz.js

```diff
@@ -0,0 +1,28 @@
+import questions from './questions.js'
+
+const askQuestion = () => {
+    const randomNumber = Math.floor(Math.random() * questions.length)
+    const { correctAnswer: _correctAnswer, ...question } = questions[randomNumber]
+    return question
+}
+
+const answerQuestion = (question, answer) => {
+    if (!question || !Object.prototype.hasOwnProperty.call(question, 'question')
+        || !answer || !/^[a-d]$/.test(answer)
+    ) {
+        return false
+    }
+    // we could check for all object keys but in this case we we assume that each question exists only once
+    const askedQuestion = questions.find(el => el.question === question.question)
+
+    if (!askedQuestion) {
+        return false
+    }
+
+    return askedQuestion.correctAnswer === answer
```

Comment:
This block could be simplified to:

```suggestion
    return askedQuestion && askedQuestion.correctAnswer === answer
```